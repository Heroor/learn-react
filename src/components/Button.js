import React from 'react'
import PropTypes from 'prop-types'

const Button = ({
  onClick,
  className,
  children
}) => (
  <button
    onClick={onClick}
    className={className}
    type="button"
  >
    {children}
  </button>
)

// 为子组件接受父组件的props限定类型
// • PropTypes.array
// • PropTypes.bool
// • PropTypes.func
// • PropTypes.number
// • PropTypes.object
// • PropTypes.string
// • PropTypes.node  // 可以为节点或字符串
// • PropTypes.element  // 元素 或 react元素

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

// 设置默认props
Button.defaultProps = {
  className: ''
}

export default Button
