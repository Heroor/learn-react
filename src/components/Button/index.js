import React from 'react'
import PropTypes from 'prop-types'

const Button = ({
  onClick,
  className,
  children,
  type
}) => (
  <button
    onClick={onClick}
    className={className}
    type={type}
  >
    {children}
  </button>
)


// 为子组件接受父组件的props限定类型
// • PropTypes.array
// • PropTypes.bool
// • PropTypes.func
// • PropTypes.number
// • PropTypes.object
// • PropTypes.string
// • PropTypes.node  // 可以为节点或字符串
// • PropTypes.element  // 元素 或 react元素

Button.propTypes = {
  onClick: PropTypes.func,
  className: PropTypes.string,
  children: PropTypes.node
}

// 设置默认props
Button.defaultProps = {
  className: '',
  children: 'button',
  type: 'button'
}

export default Button
