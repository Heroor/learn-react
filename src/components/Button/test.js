import React from 'react'
import ReactDOM from 'react-dom'
import Button from './index'

import renderComponentToSnapshot from '../../config/TestConfig'

// Button test
describe('Button', () => {
  const props = {
    onClick() {
      console.log('click!')
    },
    children: 'More'
  }
  it('render without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Button {...props}>{props.children}</Button>, div)
  })
  test('has a valid shapshot', () => {
    renderComponentToSnapshot(
      <Button {...props}>{props.children}</Button>
    )
  })
})
