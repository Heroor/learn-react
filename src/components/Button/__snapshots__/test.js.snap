// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a valid shapshot 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  More
</button>
`;
